src/hamster/session.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2020 Sébastien Granjoux <seb.sfo@free.fr>
+
+# This file is part of Project Hamster.
+
+# Project Hamster is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# Project Hamster is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with Project Hamster.  If not, see <http://www.gnu.org/licenses/>.
+
+import dbus
+import logging
+logger = logging.getLogger(__name__)
+
+class DbusSessionListener(object):
+    """Listen for GNOME manager end session event."""
+
+
+    def end_session(self):
+        """Override this method to do something at the end of a session.
+        It must not attempt to interact with the user and will be given
+        a maximum of ten seconds to perform the actions."""
+        pass
```

Comment:
Maybe this should raise `NotImplementedError`?

Reply:
I think pass is fine, there is no need to do something.
Eventually, I could extend this object to provide a hook in query_end_session_handler. I will allow to use it to delay the end of a session and it could be that nothing is needed when the session end. Anyway, it's not done currently, so I could raise `NotImplementedError` if you think it's better.